asapdiscovery-spectrum/asapdiscovery/spectrum/score_complex.py

```diff
@@ -0,0 +1,363 @@
+                    rmsd_mode="heavy",
+                    aligned_temp=aligned,
+                )
+            except Exception as e:
+                bsite_rmsd = -1
+                logging.warning("Binding site RMSD couldn't be calculated: %s", e)
+            df_save.insert(
+                loc=len(df_save.columns), column="Bsite-RMSD", value=bsite_rmsd
+            )
+            pout += f" and {bsite_rmsd} for binding site"
+        logging.info(pout)
+
+        if inputs.run_vina:
+            logging.info("Calculating the affinity score with AutoDock Vina")
+            if inputs.vina_box_x is None:
+                vina_box = None
+                logging.info("The grid box will be calculated for the complex")
+            else:
+                vina_box = [inputs.vina_box_x, inputs.vina_box_y, inputs.vina_box_z]
+
+            df_vina, out_pose = score_autodock_vina(
+                pdb_prep,
+                sdf_ligand,
+                vina_box,
+                box_size=[20, 20, 20],
+                dock=inputs.dock_vina,
+                path_to_prepare_file=str(inputs.path_to_grid_prep),
+            )
+            if out_pose is not None:
+                logging.info("Vina docking pose was successfully generated")
+                try:
+                    lig_rmsd = get_ligand_rmsd(
+                        out_pose,
+                        str(file_ref),
+                        True,
+                        rmsd_mode="oechem",
+                        overlay=False,
+                    )
+                    logging.info("The RMSD of the vina pose was: %s", lig_rmsd)
+                except Exception as e:
+                    lig_rmsd = -1
+                    logging.warning("The vina RMSD couldn't be calculated: %s", e)
+                df_vina["Vina-pose-RMSD"] = lig_rmsd
+            df_save = pd.concat([df_save, df_vina], axis=1, join="inner")
+
+        if inputs.gnina_score:
+            if inputs.gnina_script.exists():
+                logging.info("Calculating the affnity with Gnina")
+                try:
+                    df_gnina = score_gnina(
+                        f"{pdb_prep.stem}.pdb",
+                        f"{sdf_ligand.stem}.sdf",
+                        docked_prepped,
+                        inputs.gnina_out_dir,
+                        inputs.gnina_script,
+                    )
+                    logging.debug("Gnina df output is: %s", df_gnina)
+                    df_save = pd.concat([df_save, df_gnina], axis=1, join="inner")
+                except Exception as e:
+                    logging.error("The Gnina calculation failed: %s", e)
+            else:
+                logging.error(
+                    "A gnina bash script must be provided to calculate gnina scores. Won't calculate."
+                )
+        df_save.to_csv(inputs.output_csv, mode='a', index=False, header=first_write)
+        first_write = False
+
+    return
```

Comment:
If this is not returning anything then we probably don't need this line. Also, this makes me even more convinced that we probably want to have this in the workflows or cli package, as a cli directly. Related to a comment I left before.